### PR TITLE
KAFKA-17378: Fixes for performance testing

### PR DESCRIPTION
**Pull request by AndrewJSchofield**
A few small fixes as a result of investigations during performance testing.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/java/kafka/server/share/SharePartition.java [hunk 1]
```diff
@@ -351,7 +351,7 @@ public CompletableFuture<List<AcquiredRecords>> acquire(
         String memberId,
         FetchPartitionData fetchPartitionData
     ) {
-        log.trace("Received acquire request for share partition: {}-{}", memberId, fetchPartitionData);
+        log.trace("Received acquire request for share partition: {}-{}", memberId, topicIdPartition);
```
Comment (chia7712) [inline core/src/main/java/kafka/server/share/SharePartition.java]: The other logs use `groupId` and `topicIdPartition`, so I guess `memberId` should be replaced by `groupId`?

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java [hunk 2]
```diff
@@ -266,11 +266,18 @@ private Optional<UnsentRequest> maybeBuildRequest(AcknowledgeRequestState acknow
             return Optional.empty();
         } else if (!acknowledgeRequestState.maybeExpire()) {
             if (acknowledgeRequestState.canSendRequest(currentTimeMs)) {
-                acknowledgeRequestState.onSendAttempt(currentTimeMs);
-                if (onCommitAsync) {
-                    isAsyncDone.set(true);
+                UnsentRequest request = acknowledgeRequestState.buildRequest(currentTimeMs);
```
Comment (chia7712) [inline clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java]: This method has some nested if-else. Maybe we can flatten them? for example:
```java
    private Optional<UnsentRequest> maybeBuildRequest(AcknowledgeRequestState acknowledgeRequestState,
                                                      long currentTimeMs,
                                                      boolean onCommitAsync,
                                                      AtomicBoolean isAsyncDone) {
        boolean asyncDone = true;
        try {
            if (acknowledgeRequestState == null || (!acknowledgeRequestState.onClose() && acknowledgeRequestState.isEmpty())) {
                return Optional.empty();
            }

            if (acknowledgeRequestState.maybeExpire()) {
                // Fill in TimeoutException
                for (TopicIdPartition tip : acknowledgeRequestState.incompleteAcknowledgements.keySet()) {
                    metricsManager.recordFailedAcknowledgements(acknowledgeRequestState.getIncompleteAcknowledgementsCount(tip));
                    acknowledgeRequestState.handleAcknowledgeTimedOut(tip);
                }
                acknowledgeRequestState.incompleteAcknowledgements.clear();
                return Optional.empty();
            }

            if (!acknowledgeRequestState.canSendRequest(currentTimeMs)) {
                // We wait for the backoff before we can send this request.
                asyncDone = false;
                return Optional.empty();
            }

            UnsentRequest request = acknowledgeRequestState.buildRequest(currentTimeMs);
            if (request == null) {
                asyncDone = false;
                return Optional.empty();
            }

            acknowledgeRequestState.onSendAttempt(currentTimeMs);
            return Optional.of(request);
        } finally {
            if (onCommitAsync) {
                isAsyncDone.set(asyncDone);
            }
        }
    }
```

**Review by chia7712 — COMMENTED** [review]
@AndrewJSchofield overall LGTM except for some minor comments.

File: core/src/main/java/kafka/server/share/SharePartition.java (hunk 1 above)
Comment (AndrewJSchofield) [reply core/src/main/java/kafka/server/share/SharePartition.java]: Ah, yes. True. The bigger problem was that fetchPartitionData doesn't have a sensible string representation so the log line was useless. What you suggested makes total sense.

**Review by apoorvmittal10 — APPROVED** [review]
LGTM! Thanks for the fixes.

File: clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java (hunk 2 above)
Comment (apoorvmittal10) [reply clients/src/main/java/org/apache/kafka/clients/consumer/internals/ShareConsumeRequestManager.java]: Agree, flatteing is better.

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**